fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabaseRunner.java

```diff
@@ -358,15 +360,19 @@ public FDBRecordContext openContext() {
         }
 
         @SuppressWarnings("squid:S1181")
-        public CompletableFuture<T> runAsync(@Nonnull Function<? super FDBRecordContext, CompletableFuture<? extends T>> retriable) {
+        public CompletableFuture<T> runAsync(@Nonnull final Function<? super FDBRecordContext, CompletableFuture<? extends T>> retriable,
+                                             @Nonnull final BiFunction<T, Throwable, Pair<T, Throwable>> handlePostTransaction) {
```

Comment:
It's possible this should be something like `BiFunction<? super T, ? super Throwable, ? extends Pair<? extends T, ? extends Throwable>>`, but maybe that messes up using `Pair::of` as the function reference?